components/key-value/src/lib.rs

```diff
@@ -21,15 +21,21 @@ struct Component;
 impl Guest for Component {
     fn handle(request: IncomingRequest, response_out: ResponseOutparam) {
         let result = handle(request)
-            .map(|_| OutgoingResponse::new(Headers::new()))
+            .map(|r| match r {
+                Some(e) => response(500, format!("{e}").as_bytes()),
+                None => response(200, b""),
+            })
             .map_err(|e| ErrorCode::InternalError(Some(e.to_string())));
         ResponseOutparam::set(response_out, result)
     }
 }
 
-fn handle(_req: IncomingRequest) -> anyhow::Result<()> {
+fn handle(_req: IncomingRequest) -> anyhow::Result<Option<Error>> {
```

Comment:
Retro-review: this seems more surprising than `anyhow::Result<Result<(), Error>>`